src/simple/multiplex/client.rs

```diff
@@ -28,20 +28,30 @@ pub trait ClientProto<T: 'static>: 'static {
     /// Response messages.
     type Response: 'static;
 
+    /// The type of request ids to used to correlate requests to responses
+    type RequestId: RequestId;
```

Comment:
This would be a breaking change. Also, without the ability to have default associated types, this would increase the burden to get started.